pkg/ocicni/ocicni.go

```diff
@@ -222,27 +236,39 @@ func getDefaultCNINetwork(pluginDir string, cniDirs []string, vendorCNIDirPrefix
 			logrus.Warningf("CNI config list %s has no networks, skipping", confFile)
 			continue
 		}
-		logrus.Infof("CNI network %s (type=%v) is used from %s", confList.Name, confList.Plugins[0].Network.Type, confFile)
+		if confList.Name == "" {
+			confList.Name = path.Base(confFile)
+		}
+
+		logrus.Infof("Found CNI network %s (type=%v) at %s", confList.Name, confList.Plugins[0].Network.Type, confFile)
+
 		// Search for vendor-specific plugins as well as default plugins in the CNI codebase.
-		vendorDir := vendorCNIDir(vendorCNIDirPrefix, confList.Plugins[0].Network.Type)
+		vendorDir := vendorCNIDir(plugin.vendorCNIDirPrefix, confList.Plugins[0].Network.Type)
```

Comment:
Could we flatten Plugins and all the Plugin.Network.Type in the Path? Plugins[0] will still leave it selective.

Reply:
> Could we flatten Plugins and all the Plugin.Network.Type in the Path? Plugins[0] will still leave it selective.

@rajatchopra done